client/src/Movies/Movie.js

```diff
@@ -26,43 +26,23 @@ export default class Movie extends Component {
       });
   };
   // Uncomment this code when you're ready for the stretch problems
-  // componentWillReceiveProps(newProps){
-  //   if(this.props.match.params.id !== newProps.match.params.id){
-  //     this.fetchMovie(newProps.match.params.id);
-  //   }
-  // }
+  componentWillReceiveProps(newProps){
+    if(this.props.match.params.id !== newProps.match.params.id){
```

Comment:
always make sure you check your console logs when doing a life cycle hook that calls for an update, some times it can cause an infinite loop pinging a site if and doing hundreds of requests if you set your conditionals wrong.